drivers/volume/kdmp/kdmp.go

```diff
@@ -219,19 +229,133 @@ func isDataExportCompleted(status kdmpapi.ExportStatus) bool {
+			}
+			jobName := getGenericCRName(snapshotDeleteJobPrefix, vInfo.Namespace, vInfo.PersistentVolumeClaim)
+			_, err = batch.Instance().GetJob(
+				jobName,
+				vInfo.Namespace,
+			)
+			if err != nil && k8serror.IsNotFound(err) {
+				err := dataexport.CreateCredentialsSecret(secretName, backup.Spec.BackupLocation, backup.Namespace, backup.Namespace)
+				if err != nil {
+					errMsg := fmt.Sprintf("failed to create secret [%v] in namespace [%v]: %v", secretName, backup.Namespace, err)
+					log.ApplicationBackupLog(backup).Errorf("%v", errMsg)
+					return false, fmt.Errorf(errMsg)
+				}
+				_, err = driver.StartJob(
+					drivers.WithJobName(jobName),
+					drivers.WithSnapshotID(vInfo.BackupID),
+					drivers.WithNamespace(vInfo.Namespace),
+					drivers.WithSourcePVC(vInfo.PersistentVolumeClaim),
+					drivers.WithCredSecretName(secretName),
+					drivers.WithCredSecretNamespace(secretNamespace),
+				)
+				if err != nil {
+					errMsg := fmt.Sprintf("failed to start kdmp snapshot delete job for snapshot [%v] for backup [%v]: %v",
+						vInfo.BackupID, backup.Name, err)
+					log.ApplicationBackupLog(backup).Errorf("%v", errMsg)
+					return false, fmt.Errorf(errMsg)
+				}
+				return false, nil
+			}
+			jobID := kdmputils.NamespacedName(backup.Namespace, jobName)
+			jobDeleted, err := doKdmpDeleteJob(jobID, driver)
+			if err != nil {
+				return false, err
+			}
+			if jobDeleted {
+				err = core.Instance().DeleteSecret(backup.Name, backup.Namespace)
+				if err != nil && !k8serror.IsNotFound(err) {
+					errMsg := fmt.Sprintf("failed to delete secret [%s] from namespace [%v]:' %v",
+						backup.Name, backup.Namespace, err)
+					log.ApplicationBackupLog(backup).Errorf("%v", errMsg)
+					return false, fmt.Errorf(errMsg)
+				}
+			}
+			// Even if one job is in-progress we want to keep the ApplicationBackup CR
+			// in the cluster so that next reconcile event we can check if the jobs
+			// are completed and delete complete jobs.
+			if !jobDeleted {
+				return false, nil
+			}
+		}
+	}
+
+	return true, nil
+}
+
+func doKdmpDeleteJob(id string, driver drivers.Interface) (bool, error) {
+	fn := "doKdmpDeleteJob:"
+	progress, err := driver.JobStatus(id)
+	if err != nil {
+		errMsg := fmt.Errorf("failed in getting kdmp delete job [%v] status: %v", id, err)
+		logrus.Warnf("%s %v", fn, errMsg)
+		return false, errMsg
+	}
+	switch progress.State {
+	case drivers.JobStateCompleted:
+		if err := driver.DeleteJob(id); err != nil {
+			errMsg := fmt.Errorf("deletion of job [%v] failed: %v", id, err)
+			logrus.Warnf("%s %v", fn, errMsg)
+			return false, errMsg
+		}
+		return true, nil
+	case drivers.JobStateFailed:
+		errMsg := fmt.Errorf("kdmp delete job [%v] failed to execute: %v", id, err)
+		logrus.Warnf("%s %v", fn, errMsg)
+		if err := driver.DeleteJob(id); err != nil {
+			errMsg := fmt.Errorf("deletion of job [%v] failed: %v", id, err)
+			logrus.Warnf("%s %v", fn, errMsg)
+			return false, errMsg
+		}
+		return true, nil
```

Comment:
nit: Since it is a switch, and there is no statesment after the switch case, we can avoid the return in every case for success case.

Reply:
As part of cleanup, removed it